airflow/www/views.py

```diff
@@ -771,6 +773,29 @@ def _get_log(self, ti, log_filename):
 
         return log
 
+    @expose('/get_log_js')
+    @login_required
+    @wwwutils.action_logging
+    def get_log_js(self):
+        """ JavaScript endpoint for fetching logs. """
+        logging_backend = cached_logging_backend()
+        if not logging_backend:
+            raise AirflowException("Logging backend not found.")
+
+        dag_id = request.args.get('dag_id')
+        task_id = request.args.get('task_id')
+        execution_date = request.args.get('execution_date')
+        dttm = dateutil.parser.parse(execution_date)
+        try_number = request.args.get('try_number')
+        offset = request.args.get('offset')
+
+        logs = logging_backend.get_logs(dag_id=dag_id, task_id=task_id,
+            execution_date=execution_date, try_number=try_number, offset=offset)
+        next_offset = offset if not logs else logs[-1].offset
+        message = '\n'.join([log.message for log in logs])
+
+        return jsonify(message=message, next_offset=next_offset)
```

Comment:
You mean escaping characters in the returned message, which might cause decode and rendering issues?